### PR TITLE
Add reset for dynamic page

### PR DESCRIPTION
**Pull request by mhpjay422**
Why this PR?
- Currently unable to reset fields and saved on the dynamic page

This PR contains: 
- Reset button that: 
  - Deletes all custom fields and options for the current user
  - Resets and fields to default

### CODE REVIEW

**Comment by mhpjay422** [comment]
https://user-images.githubusercontent.com/12041898/233882252-dcf47562-0609-497d-baa5-57dbb7b5a673.mp4

File: pages/api/v1/reset-workspace.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const token = await getToken({
+    req: req,
+  });
+  // console.log("gSSP token", token);
+
+  if (!token?.sub) {
+    throw new Error("No session");
+  }
+
+  const prisma = prismaClient;
```
Comment (johnmosesman) [inline pages/api/v1/reset-workspace.ts:20]: I would just reference `prismaClient` directly, no need to make a separate variable.

`prismaClient.options..`

File: pages/api/v1/reset-workspace.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    },
+  });
+
+  const deleteCustomFields = await prisma.customField.deleteMany({
+    where: {
+      userId: token.sub,
+    },
+  });
+
+  res.send("Options and Custom Fields deleted");
```
Comment (johnmosesman) [inline pages/api/v1/reset-workspace.ts:34]: It's odd to return just text here, if you're not returning a flash message or something a 200 response is probably fine

File: pages/dynamic-templates.tsx [hunk 3]
```diff
@@ -239,12 +240,48 @@ const DynamicTemplates: NextPageWithLayout<Props> = ({
     }
   };
 
+  const handleResetSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    try {
+      if (
+        confirm(
+          "This will reset current adjusted field options and delete all saved data. Are your sure you want to reset?"
```
[the remark this answers is not in the document]
Comment (mhpjay422) [reply pages/dynamic-templates.tsx]: Gotcha. I'll match the verbage ("field options") thats used in the header description text

CleanShot 2023-04-24 at 17 29 27

**Pull request merged**